### PR TITLE
esp_jpeg: Added example

### PR DESCRIPTION
**Pull request by espzav**
# Checklist

- [x] Component contains License
- [x] Component contains README.md
- [ ] Component contains idf_component.yml file with `url` field defined
- [x] Component was added to [upload job](https://github.com/espressif/idf-extra-components/blob/master/.github/workflows/upload_component.yml#L18)
- [x] Component was added to [build job](https://github.com/espressif/idf-extra-components/blob/master/test_app/CMakeLists.txt#L8)
- [x] _Optional:_ Component contains unit tests
- [x] CI passing

# Change description
Moved tjpgd example from IDF to this component and used BSP for LCD initialization.

### CODE REVIEW

File: esp_jpeg/examples/tjpgd/main/pretty_effect.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/*
+ * SPDX-FileCopyrightText: 2022 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: CC0-1.0
+ */
+
+#pragma once
+#include <stdint.h>
+#include "esp_err.h"
+
```
Comment (suda-morris) [inline esp_jpeg/examples/tjpgd/main/pretty_effect.h:10]: missing cpp guard

File: esp_jpeg/examples/tjpgd/main/pretty_effect.c [hunk 2]
```diff
@@ -0,0 +1,62 @@
+/*
+ * SPDX-FileCopyrightText: 2022 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: CC0-1.0
+ */
+
+#include <math.h>
+#include "pretty_effect.h"
+#include "sdkconfig.h"
```
Comment (suda-morris) [inline esp_jpeg/examples/tjpgd/main/pretty_effect.c:9]: no CONFIG_xxx macro found in this file, maybe we can remove this header file

File: esp_jpeg/examples/tjpgd/main/lcd_tjpgd_example_main.c [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+ *
+ * SPDX-License-Identifier: CC0-1.0
+ */
+
+#include <stdio.h>
+#include "sdkconfig.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+//#include "esp_lcd_panel_io.h"
+//#include "esp_lcd_panel_vendor.h"
```
Comment (suda-morris) [inline esp_jpeg/examples/tjpgd/main/lcd_tjpgd_example_main.c:12]: shall we remove them?

File: esp_jpeg/examples/tjpgd/main/idf_component.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+dependencies:
+  idf: ">=5.0"
+  esp_jpeg: ">=1.0.2"
```
Comment (suda-morris) [inline esp_jpeg/examples/tjpgd/main/idf_component.yml:3]: shall we add "override_path" for it?

File: esp_jpeg/examples/tjpgd/main/decode_image.h [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ * SPDX-License-Identifier: CC0-1.0
+ */
+
+#pragma once
+#include <stdint.h>
+#include "esp_err.h"
+
+#define IMAGE_W 320
+#define IMAGE_H 240
+
```
Comment (suda-morris) [inline esp_jpeg/examples/tjpgd/main/decode_image.h:13]: missing cpp guard in this file

File: esp_jpeg/examples/tjpgd/main/decode_image.c [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+``Image decoder: jd_prepare failed (8)`` as an error) so make sure to save in the correct
+format if you want to use a different image file.
+*/
+
+#include "decode_image.h"
+#include "jpeg_decoder.h"
+#include "esp_log.h"
+#include "esp_check.h"
+#include <string.h>
+#include "freertos/FreeRTOS.h"
```
Comment (suda-morris) [inline esp_jpeg/examples/tjpgd/main/decode_image.c:22]:
```suggestion
#include <string.h>
#include "decode_image.h"
#include "jpeg_decoder.h"
#include "esp_log.h"
#include "esp_check.h"
#include "freertos/FreeRTOS.h"
```

**Review by suda-morris — CHANGES_REQUESTED** [review]
LGTM, left some small comments

**Comment by espzav** [comment]
@suda-morris Thank you for review.
note: The BSP is nto supported with IDF4.4 and lower. We will add support for IDF4.4 in future. Should I wait with this for support IDF4.4 or change IDF compatibility to `>=5.0`?

**Comment by suda-morris** [comment]
> @suda-morris Thank you for review. note: The BSP is nto supported with IDF4.4 and lower. We will add support for IDF4.4 in future. Should I wait with this for support IDF4.4 or change IDF compatibility to `>=5.0`?

No, we don't have to wait for that because this example's focus is not on the BSP, any working BSP, any version can do.

**Comment by espzav** [comment]
@suda-morris @tore-espressif Please, how can I disable tests for this example for IDF < 5.0 ?

**Comment by espzav** [comment]
@suda-morris please, is it ok now? Can we merge it after tests will pass?

File: esp_jpeg/examples/tjpgd/main/Kconfig.projbuild [hunk 7]
```diff
@@ -0,0 +1,17 @@
+menu "Example Configuration"
+    config EXAMPLE_LCD_SPI_8_LINE_MODE
```
Comment (suda-morris) [inline esp_jpeg/examples/tjpgd/main/Kconfig.projbuild:2]: shall we remove this since the BSP driver doesn't contain any LCD utilizes the SPI octal bus.

**Pull request merged**